Zend/zend_object_handlers.c

```diff
@@ -2117,13 +2114,10 @@ ZEND_API zend_function *zend_std_get_constructor(zend_object *zobj) /* {{{ */
 	if (constructor) {
 		if (UNEXPECTED(!(constructor->op_array.fn_flags & ZEND_ACC_PUBLIC))) {
 			zend_class_entry *scope = get_fake_or_executed_scope();
-			if (UNEXPECTED(constructor->common.scope != scope)) {
-				if (UNEXPECTED(constructor->op_array.fn_flags & ZEND_ACC_PRIVATE)
-				 || UNEXPECTED(!zend_check_protected(zend_get_function_root_class(constructor), scope))) {
-					zend_bad_constructor_call(constructor, scope);
-					zend_object_store_ctor_failed(zobj);
-					constructor = NULL;
-				}
+			if (!zend_check_method_accessible(constructor, scope)) {
```

Comment:
To solve the fn_flags reload problem, just insert an ZEND_ASSERT/ZEND_ASSUME here (preferably the former).

Reply:
Always inline didn't change anything for the non-JIT runs (and my local tests also indicated that the inlining happened).

Reply:
You mentioned:

> I hate this, the point of the method was that I don't need to think about this 😒

This doesn't seem to be better in that regard. An assert can at least remind you that you didn't check for `ZEND_ACC_PUBLIC`. Anyway, just my opinion, proceed as you like.